cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -346,8 +349,11 @@ public void updateApp(HttpRequest request, HttpResponder responder,
       throw new BadRequestException("Request body is invalid json: " + e.getMessage());
     }
 
+    boolean updateSchedules =
+      appRequest.canUpdateSchedules() == null ? appUpdateSchedules : appRequest.canUpdateSchedules();
+
     try {
-      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator());
+      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator(), updateSchedules);
```

Comment:
I see. You can pass in the appRequest.canUpdateSchedules() to the deployApp from the handlers to the lifecycleService and then the 

lifecycleService can read the cdap-site.xml and and do this
```
    boolean updateSchedules =
      appRequest.canUpdateSchedules() == null ? appUpdateSchedules : appRequest.canUpdateSchedules();
```

Which currently we are doing in 2-3 places in handler. It can be done just in the deployApp of the lifecycleService (the main deployApp which deos stuff around line 589)